src/addons/addon.cpp

```diff
@@ -375,6 +376,25 @@ Addon::~Addon() {
   disable();
 }
 
+void Addon::updateAddonState(State newState) {
+  m_state = newState;
+
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
```

Comment:
You need to load the state in the CTORto keep m_state in sync with the real value.